### PR TITLE
Use experimental samplers.

### LINKED ISSUE #54: Use the new sampler module in all the Book material
Description:
-------------
We made a decision to update the book material to use the new sampler module instead of the old one (the Gibbs notebook has a separate issue #55)

DoD:
------
- [x] The book examples are updated to use the new sampler module instead of the old one (except the Gibbs notebook)
- [x] Results in the book pages are checked against old samplers version and are meaningful.
- [x] Sampling progress printing is checked to look presentable (note: new samplers uses tqdm)
- [ ] The changes are reviewed and merged

Note:
------
- notes from @chaozg in https://github.com/CUQI-DTU/CUQI-Book/issues/40 could be relevant

### PR DESCRIPTION
**Pull request by nabriis**
Closes #54 

@amal-ghamdi could you help me pre-run the notebook as you did with the previous version of the book?

I added a pipeline step to store an artifact of the book when working with PR. See e.g. the run here 
https://github.com/CUQI-DTU/CUQI-Book/actions/runs/10800740763

Then one can download the book that was built and investigate/review.

Please feel free to commit to my branch.

### CODE REVIEW

File: chapter04/Plugins/CUQIpy-CIL/demo_ct.ipynb [hunk 1]
```diff
@@ -297,184 +356,2680 @@
...
...eWKyTquEHmmfyb0EX/wgx/01udqH+hHpwYg5FenRoC6JObbcDUijNMwevToVJl+dcZHoB6FWgnGR/DF+aC+ILQeQloJXu8ep02pm2EMCrZNbUsoZwbjhLhQTxDKn8K2fHlkzNLrJaRdYl2h3E7UCI0bNy5VdvO18FrG6WDsnDFjxqTKXEs1NTWpMtePO4dcl9Qqce38+te/TpX7Ms9U1tTzRca3yFOrmFXDEfP8z6on7I6ol4+//uu/tnvvvdduvfVWMzObOnWq7dixw5qbm23evHmlBd3W1pYKetPe3l72NUQIIYQQ70yiNB9vvPFG2RvXgAEDSm/ldXV1VlNTYy0tLaXjx44ds7Vr15btYBBCCCHEO5OoLx+33HKL3X///TZu3DibMmWKPf/887Zs2TL71Kc+ZWZ/+PyycOFCa2pqsvr6equvr7empiYbPHiwzZ07t08GIIQQQohTi6iXj3/4h3+wL37xi7ZgwQJrb2+32tpamz9/vv2///f/SucsXrzYjh49agsWLLDDhw/btGnTbM2aNVExPsz+4Dc64TvKO0aFr67Y43n1oydt55kLgNCnHLNPPNaHy/Lvfve7VJnxEhjTgNoJF2oAGFOCvvDt27enyvSlM26ImwuEx0JaB1/sDDOzvXv3psrMHeLqT3g/hWIphHQXbIt+fHcOOL/UADAmBeH1zK/C4+4csl+8luOgXoRrh+uednLHzfmlDTkH1NVQV0GYj8XV0tAm1BtRL8J76pJLLkmVX3rppVSZug7X5rQR8+VQu8K2Y5+xbjmkB4ol9JzsS21bTFuhdXkyc3exLva1p0S9fAwdOtSWL19uy5cv93assbHRGhsbe9UhIYQQQpzeKLeLEEIIIQpFLx9CCCGEKJQot0uRuHE+SJb90Xn77bJcn1WnkafWJVSXz68X8gGG6qb/mr5z6htc33gofwY1Hi+//HKqfNFFF6XKe/bsSZWpEXB964wpwX4zfg01HrQT9Sm+fB60CbUP1CNQK8EYI4Rb41070P/M2CmMIcG2OC5qRhgvw9W6cN1Sf8C6ma4hFKuDY3PnlPMTivPBcRGuXdbnts35pg5q3759qTLXNY+zb1z37hzShtRFnYh4fYIDBw5YFmJiJ4XiX5DQs8nXdt46CxKTVybrb0eeMUxi5stFXz6EEEIIUSh6+RBCCCFEoejlQwghhBCF0m81H
```
Comment (nabriis) [inline chapter04/Plugins/CUQIpy-CIL/demo_ct.ipynb:2430]: The posterior mean image with NUTS sampler in experimental looks very different.

**Comment by nabriis** [comment]
@amal-ghamdi and @jakobsj. This should be ready now. All results LGTM with new experimental samplers. No major difference.

File: chapter04/Plugins/CUQIpy-CIL/demo_ct.ipynb (hunk 1 above)
Comment (nabriis) [reply chapter04/Plugins/CUQIpy-CIL/demo_ct.ipynb]: Updated. It works fine now.

**Review by jakobsj — COMMENTED** [review]
There appear to be some noticeable changes in the fenics notebook, in terms of images and chains, not sure how significant. 

And in the donut the plots are no longer "axis equal", ie the donut is squashed.

And results and violin plots in pytorch example also changed.

Let us discuss today.

**Review by jakobsj — APPROVED** [review]
Thanks for the discussion on the updated results. Happy to see this merged as is.

**Review by amal-ghamdi — APPROVED** [review]

**Pull request merged**